### PR TITLE
Migrate from postgres to mongodb database

### PR DESCRIPTION
**Pull request by ghost**
- removed all postgres code
- fixed style for object declarations
- removed postgres libraries

### CODE REVIEW

File: src/server/TCGAadapter.js [hunk 1]
```diff
@@ -125,22 +177,33 @@ export class TCGAAdapter {
...
-      // Pass just json with gene:correlation
-      const filteredRvals = rangeFilter(results.rows[0].rvalue, min, max)
-      // Map to object with gene and rvals as keys
-      const mappedRvals = _.map(filteredRvals, (value, gene) => {
-        return { gene: gene, Rvalue: value }
-      })
-      return _.reverse(
-        _.sortBy(mappedRvals, o => {
-          return o.Rvalue // We use reciprocal to achieve descending sort
+    return this.mongodb
```
Comment (fabubaker) [inline src/server/TCGAadapter.js:202]: There's a lot of processing in this endpoint that we do outside of mongodb (`rangeFilter`, `mappedRvals`).

Is it possible to move that to mongodb using `aggregate`?

Comment (ghost) [reply src/server/TCGAadapter.js:202]: Yes, update reflected in most recent commit

File: src/server/TCGAadapter.js [hunk 2]
```diff
@@ -125,22 +163,49 @@ export class TCGAAdapter {
...
-      // Pass just json with gene:correlation
-      const filteredRvals = rangeFilter(results.rows[0].rvalue, min, max)
-      // Map to object with gene and rvals as keys
-      const mappedRvals = _.map(filteredRvals, (value, gene) => {
-        return { gene: gene, Rvalue: value }
+    return this.mongodb
+      .collection(table)
+      .aggregate(query)
+      .toArray()
+      .then(results => {
```
Comment (fabubaker) [inline src/server/TCGAadapter.js:206]: Right now the output of this query (`results`) looks something like:

```
[{"rvalues":[{"gene":"HNRNPAB","rvalue":0.468397770192776},{"gene":"SAMD14","rvalue":0.45895966910601},{"gene":"PARP12","rvalue":0.381557144877552},{"gene":"LSM4","rvalue":0.367733023644426}...
```
Can we just directly return the inner array, i.e.

```
[{"gene":"HNRNPAB","rvalue":0.468397770192776},{"gene":"SAMD14","rvalue":0.45895966910601}...
```

**Review by fabubaker — CHANGES_REQUESTED** [review]

File: src/server/MCFadapter.js [hunk 3]
```diff
@@ -2,44 +2,85 @@ import { NormalDensityZx, computeCurve } from './utility'
...
+    const limitedDataLine = this.mongodb
+      .collection('mcf_avg_psi')
+      .aggregate(query)
+      .toArray()
       .then(result => {
-        const avgPsiVals = _.map(result.rows, r => r[`${sample}_avg_log2_psi`])
+        const avgPsiVals = _.map(result, r => r[`${sample}_avg_log2_psi`])
         return computeCurve(this.binsHash, avgPsiVals, sample + '_u12_ia')
       })
     return Promise.all([fullDataLine, limitedDataLine]).then(values => {
```
Comment (fabubaker) [inline src/server/MCFadapter.js]: There's no need for the `then` if you're just returning `values`.

File: src/server/MCFadapter.js [hunk 4]
```diff
@@ -83,83 +118,88 @@ export class MCFAdapter {
...
+    }
+    const order = {
+      intron_number: 1
+    }
+    return this.mongodb
+      .collection('mcf_intron_psi')
+      .find(conditions, { projection: projection })
+      .sort(order)
+      .toArray()
       .then(results => {
```
Comment (fabubaker) [inline src/server/MCFadapter.js]: The `then` is not required, since you're just returning the `results` anyways.

File: src/server/TCGAadapter.js [hunk 5]
```diff
@@ -1,60 +1,89 @@
...
+    const u12DataLine = this.mongodb
+      .collection('tcga_brca_genes_median')
+      .aggregate(query)
+      .toArray()
       .then(result => {
-        const medianCounts = _.map(result.rows, r => {
+        const medianCounts = _.map(result, r => {
           return r.median_log2_norm_count_plus_1
         })
         return computeCurve(this.binsHash, medianCounts, sample + '_u12')
```
Comment (fabubaker) [inline src/server/TCGAadapter.js:89]: There's no need for the `then` after `Promise.all`.

**Review by fabubaker — CHANGES_REQUESTED** [review]

File: src/server/TCGAadapter.js (hunk 2 above)
Comment (ghost) [reply src/server/TCGAadapter.js:206]: I couldn't find any way to directly return the nested array

Comment (fabubaker) [reply src/server/TCGAadapter.js:206]: Ok, I see, `aggregate` always returns an Object so this isn't possible inside mongo.

**Review by fabubaker — APPROVED** [review]

**Pull request merged**